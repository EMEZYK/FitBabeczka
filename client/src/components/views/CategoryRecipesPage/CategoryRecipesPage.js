import React from "react";
import { useContext } from "react";
import { RecipesContext } from "../../../context/recipes-context";
import AllDishes from "../../ui/AllDishes/AllDishes";
import { FlexWrapper } from "../../global-styles/Flex.styled";
import FooterComponent from "../../ui/Footer/Footer";
import { PageSubtitle } from "../../ui/PageSubtitle/PageSubtitle.styled";

const CategoryRecipesPage = ({ categoryId, pageTitle, toEdit }) => {
  const recipesContext = useContext(RecipesContext);
  const recipes = recipesContext.recipes;

  const recipesByCategory =
    recipes.length !== 0
      ? recipes.filteredData.filter((recipe) => recipe.category === categoryId)
      : [];

  const renderRecipes = () => {
    if (recipesByCategory) {
      return (
        <>
          {toEdit ? (
            <AllDishes recipes={recipesByCategory} isEditable={true} />
          ) : (
            <AllDishes recipes={recipesByCategory} isEditable={false} />
          )}
        </>
      );
    } else {
      return <div>cannot load recipes</div>;
    }
  };

  return (
    <>
      <FlexWrapper>
        <PageSubtitle>{pageTitle}</PageSubtitle>
        {renderRecipes()}
        <FooterComponent />
      </FlexWrapper>
    </>
  );
};

export default CategoryRecipesPage;
