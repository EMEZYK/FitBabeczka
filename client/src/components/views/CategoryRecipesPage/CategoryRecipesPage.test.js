import { render, cleanup } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import CategoryRecipesPage from "./CategoryRecipesPage";

describe("CategoryRecipesPage components", () => {
  afterEach(cleanup);

  test("Should take a snapshot',", () => {
    const { asFragment } = render(<CategoryRecipesPage />);
    expect(asFragment(<CategoryRecipesPage />)).toMatchSnapshot();
  });

  test("Check if footer is displayed", () => {
    render(<CategoryRecipesPage />);
    const footer = screen.queryByTestId("footer");
    expect(footer).toBeInstanceOf(HTMLElement);
  });

  test("Check if container with recipes is displayed", () => {
    render(<CategoryRecipesPage />);
    const allDishesComponent = screen.queryByTestId("recipes-container");
    expect(allDishesComponent).toBeInstanceOf(HTMLElement);
  });
});
