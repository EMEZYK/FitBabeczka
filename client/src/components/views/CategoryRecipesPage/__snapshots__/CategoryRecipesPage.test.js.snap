// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryRecipesPage components Should take a snapshot', 1`] = `
<DocumentFragment>
  <div
    class="sc-papXJ eAONcb"
  >
    <h1
      class="sc-jdAMXn kWSjuy"
    />
    <div
      class="sc-bczRLJ cCDILE"
      data-testid="recipes-container"
    />
    <footer
      class="sc-ciZhAO faxpSw"
      data-testid="footer"
    >
      <p>
        ©2022 Ewelina Mężyk, Wszelkie prawa zastrzeżone
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
